src/compiler/es-next/compile-client-function.js

```diff
@@ -7,6 +7,7 @@ import { ClientFunctionAPIError } from '../../errors/runtime';
 import MESSAGE from '../../errors/runtime/message';
 
 const ANONYMOUS_FN_RE                = /^function\*?\s*\(/;
+const ES6_OBJ_METHOD_NAME_RE         = /^([A-Za-z\*\$]+)\s*\(/;
```

Comment:
It seems this regex too weak for function name, e.g. you can use '_' or numbers.